lib/python-parser/common.ts

```diff
@@ -16,3 +16,38 @@ export function specifierValidRange(
   const versionPartsLength = version.split(".").length;
   return versionPartsLength === 2 ? "^" : "~";
 }
+
+// This regex is used to extract the "semver" part from the version string.
+// See the tests for a better understanding (Also the reason why this is exported)
+export const VERSION_EXTRACTION_REGEX = /(?<VERSION>(\d+\.)*\d+).*/;
+
+/**
+ * This function was taken from a different semver library and slightly modified.
+ * If passed to Array.prototype.sort, versions will be sorted in descending order.
+ */
+export function compareVersions(version1: string, version2: string) {
+  const v1Match = VERSION_EXTRACTION_REGEX.exec(version1);
+  const v2Match = VERSION_EXTRACTION_REGEX.exec(version2);
+
+  if (v1Match === null || v2Match === null) {
+    return 0;
+  }
+
+  const v1 = v1Match
+    .groups!.VERSION.split(".")
+    .map((part) => Number.parseInt(part, 10));
+  const v2 = v2Match
+    .groups!.VERSION.split(".")
+    .map((part) => Number.parseInt(part, 10));
+
+  const max = v1.length > v2.length ? v1.length : v2.length;
```

Comment:
I would extract the following logic to its function, aiming for a small readability win:

```
function compareArrays(v1: any[], v2: any[]): number {
  const maxLength = Math.max(v1.length, v2.length);

  for (let i = 0; i < maxLength; i++) {
    const element1 = v1[i] || 0;
    const element2 = v2[i] || 0;

    if (element1 < element2) {
      return 1;
    } else if (element1 > element2) {
      return -1;
    }
  }

  return 0;
}
```

The `any` type should be concrete, just used it for the sake of the example.

What do you say?

Reply:
I can do that. To me the function is readable but that's subjective.

Regarding the concrete type, the right answer would be `Array[Comparable]` but I don't think Typescript has a `Comparable` type. I'll stick with `number` or `string` for now as that's what's likely to appear.